### PR TITLE
Support AutoName for Plugin Framework based resources

### PR DESCRIPTION
**Pull request by t0yv0**
This is on top of https://github.com/pulumi/pulumi-terraform-bridge/pull/1321

Fixes https://github.com/pulumi/pulumi-aws/issues/2644

Auto-naming is now supported for PF so besides fixing the warning for requested resources it rolls it out to all of these:

- resource:aws_appconfig_environment
- resource:aws_auditmanager_assessment
- resource:aws_auditmanager_assessment_report
- resource:aws_auditmanager_control
- resource:aws_auditmanager_framework
- resource:aws_cognito_user_pool_client
- resource:aws_opensearchserverless_access_policy
- resource:aws_opensearchserverless_collection
- resource:aws_opensearchserverless_security_config
- resource:aws_opensearchserverless_security_policy
- resource:aws_opensearchserverless_vpc_endpoint
- resource:aws_quicksight_vpc_connection
- resource:aws_resourceexplorer2_view
- resource:aws_route53_cidr_collection
- resource:aws_route53_cidr_location
- resource:aws_simpledb_domain

TODOs:

- [x] release underlying bridge support
- [x] verify end to end on one of these resources
- [x] rebuild schema and SDKs

### CODE REVIEW

**Comment by t0yv0** [comment]
Rebuilding SDK demonstrates schema changes, the "name" parameter becomes optional:

```
sdk/nodejs/appintegrations/dataIntegration.ts
sdk/nodejs/auditmanager/assessment.ts
sdk/nodejs/auditmanager/assessmentReport.ts
sdk/nodejs/auditmanager/control.ts
sdk/nodejs/auditmanager/framework.ts
sdk/nodejs/auditmanager/getControl.ts
sdk/nodejs/chime/sdkvoiceSipMediaApplication.ts
sdk/nodejs/chime/sdkvoiceSipRule.ts
sdk/nodejs/chime/sdkvoiceVoiceProfileDomain.ts
sdk/nodejs/cleanrooms/collaboration.ts
sdk/nodejs/cloudfront/fieldLevelEncryptionProfile.ts
sdk/nodejs/cloudfront/keyGroup.ts
sdk/nodejs/cloudfront/publicKey.ts
sdk/nodejs/emrcontainers/jobTemplate.ts
sdk/nodejs/finspace/kxCluster.ts
sdk/nodejs/finspace/kxDatabase.ts
sdk/nodejs/finspace/kxEnvironment.ts
sdk/nodejs/finspace/kxUser.ts
sdk/nodejs/globalaccelerator/customRoutingAccelerator.ts
sdk/nodejs/globalaccelerator/customRoutingListener.ts
sdk/nodejs/glue/dataQualityRuleset.ts
sdk/nodejs/opensearch/serverlessAccessPolicy.ts
sdk/nodejs/opensearch/serverlessCollection.ts
sdk/nodejs/opensearch/serverlessSecurityConfig.ts
sdk/nodejs/opensearch/serverlessSecurityPolicy.ts
sdk/nodejs/opensearch/serverlessVpcEndpoint.ts
sdk/nodejs/quicksight/theme.ts
sdk/nodejs/quicksight/vpcConnection.ts
sdk/nodejs/resourceexplorer/view.ts
sdk/nodejs/route53/cidrCollection.ts
sdk/nodejs/route53/cidrLocation.ts
sdk/nodejs/sfn/alias.ts
```

**Comment by t0yv0** [comment]
Verified on aws:route53/cidrCollection:CidrCollection

File: examples/plugin-framework/Pulumi.yaml [hunk 1]
```diff
@@ -5,12 +5,18 @@ outputs: {}
...
-    properties:
-      type: LOCAL
+
+  # It turns out that resourceexplorer can only be provisioned once per region, creating conflicts if this test run in
+  # parallel. Perhaps not a great example.
+  #
+  # resourceexplorer-example:
+  #   type: aws:resourceexplorer/index:Index
+  #   properties:
+  #     type: LOCAL
```
Comment (iwahbe) [inline examples/plugin-framework/Pulumi.yaml:15]: Can we just remove this. We could also have an example that isn't tested.

File: provider/go.mod [hunk 2]
```diff
@@ -19,6 +19,9 @@ replace (
 	github.com/hashicorp/terraform-provider-aws => ../upstream
 	github.com/hashicorp/terraform-provider-aws/shim => ./shim
 	github.com/hashicorp/vault => github.com/hashicorp/vault v1.2.0
+
+	github.com/pulumi/pulumi-terraform-bridge/pf => github.com/pulumi/pulumi-terraform-bridge/pf v0.14.2-0.20230802145745-14057592edb9
+	github.com/pulumi/pulumi-terraform-bridge/v3 => github.com/pulumi/pulumi-terraform-bridge/v3 v3.55.1-0.20230802145745-14057592edb9
```
Comment (iwahbe) [inline provider/go.mod:24]: Can we set these versions as a `require` instead of as a `replace`. That way it will be updated normally after merging.

**Review by iwahbe — COMMENTED** [review]
Two nits, but otherwise looks good.

**Comment by t0yv0** [comment]
```
command.go:83: Invoke '/opt/hostedtoolcache/pulumi/3.76.1/x64/pulumi login --cloud-url https://api.pulumi-staging.io/' failed: exit status 255
  Logging in using access token from PULUMI_ACCESS_TOKEN
  error: problem logging in: invalid access token
```

Hmm, is this a flake or something more serious? Trying again.

**Comment by t0yv0** [comment]
No it's persistent. Will investigate tomorrow.

**Comment by iwahbe** [comment]
@t0yv0 Could this be related to https://pulumi.slack.com/archives/CBUP505U7/p1691101512583479?

**Review by iwahbe — APPROVED** [review]
LGTM

**Comment by mikhailshilkov** [comment]
Indeed, I think the test failures were for an unrelated reason (org tokens). I reran and it's green now.

**Comment by t0yv0** [comment]
thank you!

**Pull request merged**